include/admin-switcher.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+
+class PLL_Admin_Switcher extends PLL_Switcher {
+
+	/**
+	 * Get the language elements for use in a walker
+	 *
+	 * @since 1.2
+	 *
+	 * @param PLL_Frontend_Links $links Instance of PLL_Frontend_Links.
+	 * @param array              $args  Arguments passed to {@see PLL_Switcher::the_languages()}.
+	 * @return array Language switcher elements.
+	 */
+	protected function get_elements( $links, $args ) {
+		$first = true;
+		$out   = array();
+
+		foreach ( $links->model->get_languages_list( array( 'hide_empty' => $args['hide_if_empty'] ) ) as $language ) {
```

Comment:
There is two cases here:

- `PLL_Admin_Switcher` will never hide an empty language (see line 69).
- `PLL_Frontend_Switcher` would also like to exclude language for posts with no translations

So we could extract `$links->model->get_languages_list( array( 'hide_empty' => $args['hide_if_empty'] )` into a method that will change between `PLL_Admin_Switcher` and `PLL_Frontend_Switcher`.

Read the following comments to see how it could change the code.